From 9626d48b51131a900065b8e5158947b6efc9f493 Mon Sep 17 00:00:00 2001
From: "Axel Garcia K." <git@axelgk.com>
Date: Tue, 5 May 2026 18:57:45 +0000
Subject: [PATCH 1/5] Gate qpsx MIPS32r2 GPU code

Make the MIPS32r2 GPU inner loop opt-in so SF2000 builds can stay on the MIPS32 ISA while preserving the optimized path for builds that explicitly enable it.
---
 Makefile.libretro | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/Makefile.libretro b/Makefile.libretro
index 9a0e4b5..a3842f1 100644
--- a/Makefile.libretro
+++ b/Makefile.libretro
@@ -90,6 +90,10 @@ endif
 # XA audio hack
 MORE_CFLAGS += -DXA_HACK
 
+ifeq ($(QPSX_ENABLE_MIPS32R2), 1)
+MORE_CFLAGS += -DQPSX_ENABLE_MIPS32R2=1
+endif
+
 CFLAGS  += $(MORE_CFLAGS)
 CXXFLAGS = $(CFLAGS)
 
@@ -134,9 +138,12 @@ OBJS += \
 # QPSX v099: Added psxmem_asm.o for memory write optimization
 ifeq ($(platform), sf2000)
 OBJS += \
-	src/gpu/$(GPU)/gpu_inner_mips32.o \
 	src/gte_asm.o \
 	src/psxmem_asm.o
+ifeq ($(QPSX_ENABLE_MIPS32R2), 1)
+OBJS += \
+	src/gpu/$(GPU)/gpu_inner_mips32.o
+endif
 endif
 
 # SPU - pcsxrearmed with libretro audio backend
